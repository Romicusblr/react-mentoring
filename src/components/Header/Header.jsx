import React from 'react';
import { Switch, Route, Link } from 'react-router-dom';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import BackToMain from '../BackToMain';
import Logo from '../Logo';
import style from './Header.module.css';

const Header = ({ className, children }) => (
  <section
    className={classNames(style.header, className)}
  >
    <header>
      <Logo />
      <Switch>
        <Route path="/film/">
          <Link to="/search">
            <BackToMain />
          </Link>
        </Route>
      </Switch>
    </header>
    {children}
  </section>
);

Header.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node,
};

Header.defaultProps = {
  className: '',
  children: null,
};

export default Header;
