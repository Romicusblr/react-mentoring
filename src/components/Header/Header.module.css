.header {
  position: relative;
  padding: 0 var(--padding-header);
  min-height: 30rem;
  display: flex;
  flex-direction: column;
  background-image: url('./bg.png');
  &>header {
    display: flex;
    justify-content: space-between;
    align-items: center;
  }
  &>h3 {
    margin: 0;
    position: absolute;
    top: 1.5rem;
    left: 4rem;
  }
}
