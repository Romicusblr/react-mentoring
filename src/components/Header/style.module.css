.header {
  position: relative;
  padding: 5rem var(--padding-header);
  display: flex;
  flex-direction: column;
  background-image: url('./bg.png');

  &>h1 {
    text-transform: uppercase;
    margin-bottom: 3rem;
  }

  &>h3 {
    margin: 0;
    position: absolute;
    top: 1.5rem;
    left: 4rem;
  }
}
