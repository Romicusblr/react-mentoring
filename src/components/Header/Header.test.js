import Header from '.';

fdescribe('Header', () => {
  it('should be selectable by class "header"', () => {
    expect(shallow(<Header />).is('.header')).toBe(true);
  });

  it('should match to snapshot', () => {
    const snap = shallow(<Header />);
    expect(snap).toMatchSnapshot();
  });
});
