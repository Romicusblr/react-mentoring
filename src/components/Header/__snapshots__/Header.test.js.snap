// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match to snapshot 1`] = `
<header
  className="header"
>
  <Logo
    className=""
  />
  <h1>
    find your movie
  </h1>
  <SearchBar
    className=""
  />
</header>
`;
